### ISSUE TITLE
Handle categorical data

### ISSUE THREAD

**Issue by dlebauer** [terraref/traitvis-webapp#11]
Violin plots do not do a good job of visualizing categorical data.  For example, these are all integer values between 0 and 10

![image](https://user-images.githubusercontent.com/464871/32520648-4fc57dda-c3d6-11e7-9797-2f3df2f891d6.png)

The violin plot is bumpy

For any data with <= 15 categories, it would be great to replace the violins with bar charts, but I can't find a straightforward way to do this. 

However, something like a dotplot overlaid on the boxplot might work:

```r
ggplot(data = plot_data) + 
  geom_boxplot(data = plot_data, 
               aes(x = as.Date(date), y = mean2, ymin = min(mean2), ymax = max(mean2),
                   group = as.Date(date)),
               outlier.alpha = 0.25, width = 0.1) +
  geom_dotplot(scale = 'width', width = 0.5,
               aes(x = as.Date(date), y = mean2, 
                   group = as.Date(date)),
               binaxis='y', stackdir='up',
               stackratio=0.5, dotsize=0.1) 
```

![image](https://user-images.githubusercontent.com/464871/32521270-82425830-c3d8-11e7-895c-c0511140b9e1.png)

**Comment by dlebauer** [comment]
Make sure to test cases where the number of points in each category varies ... the parameters stackratio and dotsize could be modified to adjust for this. Also experiment with ` position = position_jitter(height = 0.1)` to spread the points so that each one is visible.

**Comment by dlebauer** [comment]
resolved by #12
